packages/@uppy/dashboard/src/Dashboard.jsx

```diff
@@ -705,6 +705,24 @@ export default class Dashboard extends UIPlugin {
     this.uppy.emit('restore-canceled')
   }
 
+  #generateLargeThumbnailIfSingleFile = () => {
+    if (this.opts.disableThumbnailGenerator) {
+      return
+    }
+
+    const LARGE_THUMBNAIL = 600
+    const files = this.uppy.getFiles()
+
+    if (files.length === 1) {
+      const thumbnailGenerator = this.uppy.getPlugin(`${this.id}:ThumbnailGenerator`)
+      thumbnailGenerator?.setOptions({ thumbnailWidth: LARGE_THUMBNAIL })
+      const fileForThumbnail = { ...files[0], preview: undefined }
+      thumbnailGenerator.requestThumbnail(fileForThumbnail).then(() => {
+        thumbnailGenerator?.setOptions({ thumbnailWidth: this.opts.thumbnailWidth })
+      })
+    }
+  }
+
```

Comment:
The need for `const LARGE_THUMBNAIL = 600` is not evident for all images, so here is one example where it does matter.

Without `#generateLargeThumbnailIfSingleFile`:

<img width="645" alt="Screenshot 2024-07-20 at 00 06 16" src="https://github.com/user-attachments/assets/27e2d1d3-4dbc-4932-9d33-05e08b97a6ff">

With`#generateLargeThumbnailIfSingleFile`:

<img width="645" alt="Screenshot 2024-07-20 at 00 05 43" src="https://github.com/user-attachments/assets/1410fe12-0051-43f8-8944-5018896b1466">